web/src/hashchange.js

```diff
@@ -58,6 +58,13 @@ function show_all_message_view() {
     narrow.deactivate();
 }
 
+function is_somebody_else_profile_open() {
+    return (
+        user_profile.get_user_id_if_user_profile_modal_open() !== undefined &&
+        !people.is_my_user_id(user_profile.get_user_id_if_user_profile_modal_open())
```

Comment:
Can we instead do `user_profile.get_user_id_if_user_profile_modal_open() === my_current_user_id` here?

Reply:
Like this? I am doing that first check to make sure a profile is open.
```js
    return (
        user_profile.get_user_id_if_user_profile_modal_open() !== undefined &&
        user_profile.get_user_id_if_user_profile_modal_open() !== people.my_current_user_id()
    );
```

Reply:
So, `people.my_current_user_id` will not be undefined in any case except spectators and we anyways do not allow spectators to access stream or groups url. So we can directly compare it like described in the above comment, but we may use this function in future for spectators in some case and so we can keep it as it is for now.

Reply:
I got confused, the latest change in the PR is fine, no need to do any more changes for this.